example/oven_fsm.dart

```diff
@@ -0,0 +1,228 @@
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.red().value,
+            counterReset < 0,
+            en < 0,
+          ]),
+
+      // identifier: completed state, represent by `OvenState.completed`.
+      State<OvenState>(OvenState.completed,
+          // events:
+          // When the button `start` is pressed during completed state,
+          // OvenState will changed to `OvenState.cooking` state.
+          events: {
+            Logic(name: 'button_start')..gets(button.eq(Button.start())):
+                OvenState.cooking
+          },
+          // actions:
+          // During the start state, `led` is change to green; timer's
+          // `counterReset` is set to 1 (Reset value);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.green().value,
+            counterReset < 1,
+            en < 0,
+          ])
+    ];
+
+    // Assign the oven StateMachine object to public variable declared.
+    oven = StateMachine<OvenState>(clk, reset, OvenState.standby, states);
+  }
+
+  // An ovenStateMachine that represent in getter.
+  StateMachine<OvenState> get ovenStateMachine => oven;
+}
+
+Future<void> main() async {
+  // Signals `button` and `reset` that mimic user's behaviour of button pressed
+  // and reset.
+  //
+  // Width of button is 2 because button is represent by 2-bits signal.
+  final button = Logic(name: 'button', width: 2);
+  final reset = Logic(name: 'reset');
+
+  // Build an Oven Module and passed the `button` and `reset`.
+  final oven = OvenModule(button, reset);
+
+  // Generate a Mermaid FSM diagram and save as the name `oven_fsm.md`.
+  // Note that the extension of the files is recommend as .md or .mmd.
+  //
+  // Check on https://mermaid.js.org/intro/ to view the diagram generated.
+  // If you are using vscode, you can download the mermaid extension.
+  oven.ovenStateMachine.generateDiagram(outputPath: 'oven_fsm.md');
+
+  // Before we can simulate or generate code with the counter, we need
+  // to build it.
+  await oven.build();
+
+  // Now let's try simulating!
+
+  // Let's start off with asserting reset to Oven.
+  reset.inject(1);
+
+  // Attach a waveform dumper so we can see what happens.
+  WaveDumper(oven, outputPath: 'oven.vcd');
+
+  // Drop reset at time 25.
+  Simulator.registerAction(25, () => reset.put(0));
+
+  // Press button start => `00` at time 25.
+  Simulator.registerAction(25, () {
+    button.put(Button.start().value);
+  });
+
+  // Press button pause => `01` at time 50.
+  Simulator.registerAction(50, () {
+    button.put(Button.pause().value);
+  });
+
+  // Press button resume => `10` at time 70.
+  Simulator.registerAction(70, () {
+    button.put(Button.resume().value);
+  });
+
+  // Print a message when we're done with the simulation!
+  Simulator.registerAction(120, () {
+    print('Simulation End');
```

Comment:
If this example is added to `test/example_test.dart`, then `print` must be wrapped in `if (!noPrint)` like the other examples.